### PR TITLE
Fix Opt injection

### PR DESCRIPTION
**Pull request by RezaYazdaniAminabadi**
This PR fixes an issue that caused for OPT models only due to the recent policy refactoring. 
It also adds some check at inference-test to verify injection of our kernels in the model

cc: @arashashari @mrwyattii @jeffra

### CODE REVIEW

File: deepspeed/inference/engine.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def __init__(self, model, config):
         # retain this from the old conditional argument being passed to apply_injection_policy()
         if not config.replace_with_kernel_inject:
             config.checkpoint = None
-
+        import pdb;pdb.set_trace()
```
Comment (arashb) [inline deepspeed/inference/engine.py:114]: debugging code left by mistake here.

Comment (RezaYazdaniAminabadi) [reply deepspeed/inference/engine.py:114]: I am removing it now, thanks @arashb

File: tests/unit/inference/test_inference.py [hunk 2]
```diff
@@ -213,6 +215,17 @@ def assert_fn(model_w_task):
         NotImplementedError(f'assert_fn for task "{task}" is not implemented')
     return assert_fn
 
+@pytest.fixture
+def check_injection(model):
+    def verify_injection(module):
+        for child in module.children():
+            if isinstance(child, nn.ModuleList):
+                assert isinstance(child[0], DeepSpeedTransformerInference),
```
Comment (arashb) [inline tests/unit/inference/test_inference.py:223]: here seems to be syntax mistake. I think you would need `\` at the end of the line.

Comment (RezaYazdaniAminabadi) [reply tests/unit/inference/test_inference.py:223]: yes, corrected now

**Review by mrwyattii — APPROVED** [review]
LGTM

**Review by awan-10 — APPROVED** [review]
Looks good.

**Comment by lanking520** [comment]
Still saw test failing, any updates in here?

**Comment by aws-stdun** [comment]
Any ETA on merging this? Would like to use this fix.

**Comment by RezaYazdaniAminabadi** [comment]
Hi @lanking520 and @aws-stdun 

Sorry for my delay here. I have pushed a fix to pass the tests. Will merge soon.

Best,
Reza

**Comment by aws-stdun** [comment]
This PR fixed a few things, but underlying problems remain that prevent practical use.
I've filed another bug report with more details:
https://github.com/microsoft/DeepSpeed/issues/2747

**Pull request merged**